### PR TITLE
Fix logic of error string match in `bad-ssts` (#12049)

### LINKED ISSUE #12329: tikv-ctl: error string match in `bad-ssts` tool
### What version of TiKV are you using?

master

### What operating system and CPU are you using?

Don't care

### Steps to reproduce

just modify a normal sst file to bad file and execute `bad-ssts` to verify and cannot get info as expected because the sst file name is not recognized.

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #12049 to release-5.3
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/12149
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/12149:release-5.3-e1fae9469b33
```

---

### What is changed and how it works?

close #12329

https://github.com/tikv/tikv/pull/10259 introduced `bad-ssts` feature to scan for damaged sst info. This PR fixed handling logic of error info returned from rocksdb.

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

Side effects

No

### Release note 

```release-note
Fix logic of error string match in `bad-ssts`.
```

### CODE REVIEW

File: cmd/tikv-ctl/src/cmd.rs [hunk 1]
```diff
@@ -0,0 +1,690 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
```
Comment (tabokie) [inline cmd/tikv-ctl/src/cmd.rs:1]: This file should be removed?

**Review by tabokie — APPROVED** [review]

File: cmd/tikv-ctl/src/cmd.rs [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+    )]
+    /// Encode a key in escaped format
+    pub encode: Option<String>,
+
+    #[structopt(subcommand)]
+    pub cmd: Option<Cmd>,
+}
+
+#[derive(StructOpt)]
+pub enum Cmd {
```
Comment (Connor1996) [inline cmd/tikv-ctl/src/cmd.rs:106]: why include these changes?

**Comment by glorv** [comment]
@Connor1996 The merge conflicts is resolved, PTAL

**Review by glorv — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**